autocertkit/ack_cli.py

```diff
@@ -170,79 +171,79 @@ def kvp_string_to_rec(string):
 
 
 def parse_netconf_file(filename):
-
-    arrs = utils.read_valid_lines(filename)
-    utils.log.debug("netconf_file: '%s'" % arrs)
+    """Parse network config file in ini format
+    E.g.
+        [eth0]
+        network_id = 0
+        vlan_ids = 200,204,240
+        vf_driver_name = igbvf
+        vf_driver_pkg = igbvf-2.3.9.6-1.x86_64.rpm
+
+        [static_0_200]
+        ip_start = 192.168.0.2
+        ip_end = 192.168.0.10
+        netmask = 255.255.255.0
+        gw = 192.168.0.1
+
+        [static_management]
+        # similar to static_0_200
+    """
+    utils.log.debug("Parse network config file: %s" % filename)
+
+    cp = ConfigParser.ConfigParser()
+    cp.read(filename)
     rec = {}
-    for arr in arrs:
-
-        if arr.count('=') != 1:
-            raise Exception(
-                "Error: format of netconf file should be 'key = value'")
-
-        key, value = [items.strip() for items in arr.split('=')]
-
-        if key.startswith('eth'):
+    for section in cp.sections():
+        if section.startswith('eth'):
             # Ethernet Interface
-            utils.log.debug("Ethernet Interface: '%s'" % key)
-
-            # Each line is in the format:
-            # ethX = 3,[235,236,237],igbvf,igbvf-2.3.9.6-1.x86_64.rpm # comment
-
-            # remove comment
-            value = value.split('#')[0].strip()
-
-            # preprocess vlan id field
-            try:
-                # Extract bracketed substring of vlan ids
-                vlan_str = value[value.index('['):value.index(']')+1]
-            except ValueError:
-                raise utils.InvalidArgument('VLAN IDs', value, 'Not found')
-            vlan_str_new = vlan_str.replace(',', ';')
-            value = value.replace(vlan_str, vlan_str_new[1:-1])
-
-            # now value: 3,235;236;237,igbvf,igbvf-2.3.9.6-1.x86_64.rpm
-            csv = [v.strip() for v in value.split(',')]
-            if len(csv) != 4:
-                raise utils.InvalidArgument(
-                    """Lack of argument "%s", should be in format: <network_id>,[<vlan_ids>],<vf_driver_name>,<vf_driver_pkg>"""
-                    % value)
+            utils.log.debug("Ethernet Interface: '%s'" % section)
 
             # Network ID is a label of the physical network the adapter has been connected to
             # and should be uniform across all adapters.
-            utils.log.debug("Network IDs: '%s'" % csv[0])
-            network_id = int(csv[0])
-
-            # Convert values to integers
-            vlan_ids = [int(x) for x in csv[1].split(';')]
+            network_id = cp.get(section, 'network_id')
```

Comment:
I think you can use `getint` to cast and validate the config file did specify an `int` here?

Reply:
Yes getint is able to, while I want to get the original value and log it, it will bring convenience to locate root cause of problem.
I added exception catch here.